resources/js/components/ImportCSVModal.vue

```diff
@@ -0,0 +1,105 @@
+<template>
+    <Modal title="Import Balance Entries">
+        <div id="csv-import">
+            <div class="px-5 py-12 border-t-2 border-b-2 border-gray-300">
+                <Label label=".CSV File" />
+                <input class="hidden" type="file" @change="setSelectedFile" ref="importer">
+
+                <div class="border-2 border-gray-300 p-4 flex flex-row rounded-md justify-between">
+                    <p>{{ file ? file.name : 'No file chosen' }}</p>
+                    <Link name="Select File" @onClick="$refs.importer.click()" />
+                </div>
+            </div>
+
+            <SaveActions
+                :can-submit="canSubmit"
+                :error="errorMessage"
+                save-btn="Import"
+                :show-loader="loading"
+                @closeAction="closeModal($event)"
+            />
+        </div>
+    </Modal>
+</template>
+
+<script>
+  import Label from './Label';
+  import Link from './Link';
+  import SaveActions from './SaveActions';
+  import { SET_ROW_COUNT, RESET_ENTRY, DISABLE_ENTRIES, SET_IMPORT_ALERT } from '../constants';
+  import Modal from './Modal';
+
+  export default {
+    components: {
+      Label,
+      Link,
+      Modal,
+      SaveActions,
+    },
+
+    computed: {
+      canSubmit: function () {
+        return !this.loading && this.file && this.acceptedTypes.includes(this.file.type);
```

Comment:
If I refreshed the page `this.loading` would be false but the process would still be happening in the background.